### PR TITLE
Fix git commit not being available in dist build

### PR DESCRIPTION
**Pull request by xPaw**
"temporary" fix, a proper fix would be #4664.

### CODE REVIEW

File: server/helper.ts [hunk 1]
```diff
@@ -44,14 +44,27 @@ function getVersionNumber() {
...
+
+	// Typescript build will be in dist/ folder,
+	// but it will not be when running `dev`.
+	if (rootPath.endsWith(path.join("dist", "server"))) {
+		rootPath = path.basename(rootPath);
+	}
+
+	// Check that .git folder exists in the expected place,
+	// to avoid git command mistakenly recursing outside of our root folder
+	if (!fs.existsSync(path.resolve(rootPath, "..", ".git"))) {
```
Comment (brunnre8) [inline server/helper.ts:67]: Note that this isn't necessarily a folder... so that comment is wrong.
If it's a worktree, .git is a file.

Comment (xPaw) [reply server/helper.ts:67]: TIL, but that doesn't strictly matter.

Comment (brunnre8) [reply server/helper.ts:67]: I know, but I already had to patch out the git folder detection once. And I'd like to avoid having to do so again so I'd appreciate it if we can get rid of the notion that .git is a folder

Comment (brunnre8) [reply server/helper.ts:67]: thanks

File: server/helper.ts [hunk 2]
```diff
@@ -44,14 +44,27 @@ function getVersionNumber() {
...
+
+	// Typescript build will be in dist/ folder,
+	// but it will not be when running `dev`.
+	if (rootPath.endsWith(path.join("dist", "server"))) {
+		rootPath = path.basename(rootPath);
+	}
+
+	// Check that .git folder exists (or a file in case of a worktree) in the expected place,
+	// to avoid git command mistakenly recursing outside of our root folder
+	if (!fs.existsSync(path.resolve(rootPath, "..", ".git"))) {
```
Comment (brunnre8) [inline server/helper.ts:67]: So looking at this a bit more closely.. why are you doing the path munging dance there?

Forgive me my ignorance, it took me a while to wrap my head around how it works.

Can't this just be simplified to

```js
function getGitCommit() {
	if (_fetchedGitCommit) {
		return _gitCommit;
	}

	_fetchedGitCommit = true;

	// Check that .git folder exists (or a file in case of a worktree) in the expected place,
	// to avoid git command mistakenly recursing outside of our root folder
	if (!fs.existsSync(path.resolve(process.cwd(), ".git"))) {
		_gitCommit = null;
		return null;
	}

	try {
		// eslint-disable-next-line @typescript-eslint/no-var-requires
		_gitCommit = require("child_process")
			.execSync(
				"git rev-parse --short HEAD", // Returns hash of current commit
				{stdio: ["ignore", "pipe", "ignore"]}
			)
			.toString()
			.trim();
		return _gitCommit;
	} catch (e: any) {
		// Not a git repository or git is not installed
		_gitCommit = null;
		return null;
	}
}
```

The whole __dirname + relative dir dance you do there only works if cwd() is set to the repo root anyhow.

Consider this:
__dirname does not point to dist server, say I run `yarn dev` in whatever dir (something seems to set cwd to the repo root no matter where you are in).
So you don't munge the path but simply head up to the repo root. That makes perfect sense to me.

For me, that essentially looks like that
```
rootpath: /home/reto/sourcecode/thelounge.hotfix/server
cwd: /home/reto/sourcecode/thelounge.hotfix
looking in /home/reto/sourcecode/thelounge.hotfix/.git
```

However, if you build the thing and start it via index.js, not yarn, you munge the rootpath to "server"
Meaning your path resolve walks down some random directory, based on cwd() only to immediately walk back up again as you do `server/../.git`

With the same location as the yarn dev example above:

```
rootpath: /home/reto/sourcecode/thelounge.hotfix/dist/server
rootpath munged to: server
cwd: /home/reto/sourcecode/thelounge.hotfix
looking in /home/reto/sourcecode/thelounge.hotfix/.git
```

In other words, that only works because the cwd() is set to index.js in the first place, and that's done by the index.js entrypoint

```js
#!/usr/bin/env node

  "use strict";

  process.chdir(__dirname);
```

Or by yarn or whatever is changing the working directory when you execute yarn dev, so that whole dance you do there seems pointless.

What am I missing?

Comment (xPaw) [reply server/helper.ts:67]: __dirname should point to the helper.js location, and looks like I indeed made a mistake with basename there.

Considering we chdir to index.js, your idea to use cwd should work regardless of dev/dist modes.

Comment (xPaw) [reply server/helper.ts]: I simplified it by using `git --git-dir ".git"`

> Specifying the location of the ".git" directory using this option (or GIT_DIR environment variable) turns off the repository discovery that tries to find a directory with ".git" subdirectory (which is how the repository and the top-level of the working tree are discovered), and tells Git that you are at the top level of the working tree.

**Comment by brunnre8** [comment]
LGTM, thanks
